scripts/fast-reboot

```diff
@@ -456,10 +471,21 @@ if [[ "$REBOOT_TYPE" = "warm-reboot" || "$REBOOT_TYPE" = "fastfast-reboot" ]]; t
     fi
 fi
 
-# We are fully committed to reboot from this point on becasue critical
+# We are fully committed to reboot from this point on because critical
 # service will go down and we cannot recover from it.
 set +e
 
+# disable trap-handlers which were set before
+trap '' EXIT HUP INT QUIT TERM KILL ABRT ALRM
+
+# "systemctl stop <service>" is expected to prevent service/containers from
+# restarting automatically. However, in some rare cases, systemctl stop doesn't
+# work as expected, and services can still auto-restart after RestartSec timer expires
+# Therefore, as a preventive measure, explicitly disable service auto-restart in the shutdown path.
+debug "Disabling auto-restart for services ..."
+grep -l "Restart=always" /usr/lib/systemd/system/*.service | xargs sed -i -e "s/\<Restart=always\>/Restart=no/"
```

Comment:
Do we have any service left in /etc/system/systemd/?

Reply:
Yes. Below three services remain untouched. I think they shouldn't be changed. Note that these services are set as  `Restart=on-failure` and not set as `Restart=always`.

`/etc/systemd/system/`:
./sshd.service:Restart=on-failure
./syslog.service:Restart=on-failure

`/usr/lib/systemd/system/` :
/usr/lib/systemd/system/
core_uploader.service:Restart=on-failure